components/support/migration/src/main/java/mozilla/components/support/migration/FennecMigrator.kt

```diff
@@ -462,10 +463,14 @@ class FennecMigrator private constructor(
                     MigrationRun(versionedMigration.version, true)
                 }
             }
+
+            // Save result of this migration immediately, so that we keep it even if we crash later
+            // in the process and do not rerun this migration version.
+            migrationStore.setOrUpdate(mapOf(versionedMigration.migration to migrationRun))
```

Comment:
Double checked, and `migrationStore` will do the right thing here - it'll augment what's currently cached, and not just overwrite existing values.

Reply:
Yeah, checked and tested that too.. :)